packages/SwingSet/src/liveslots/watchedPromises.js

```diff
@@ -116,10 +121,10 @@ export function makeWatchedPromiseManager(
           const [watcher, ...args] = watch;
           void Promise.resolve().then(() => {
             if (watcher.onRejected) {
-              watcher.onRejected(VatUpgradedMessage, ...args);
+              watcher.onRejected(disconnectObjectCapData, ...args);
```

Comment:
This definitely needs a `m.unserialize(disconnectObjectCapData)`, because we should not be passing capdata to the userspace rejection handler. I'm a little surprised that a unit test didn't catch this.. oh, I see, the unit test was updated to look for the wrong thing.

So yeah, change both this and the `test-promise-watcher.js` to look for an object literal (the disconnection object), not the capdata encoding of that object literal.

Reply:
Too many layers...

Reply:
Marginally tricky because the promise watcher wasn't passed the unserialize function, but easily taken care of...